Add React to an Existing Project (#12)
reactjs/cs.react.dev: src/content/learn/add-react-to-an-existing-project.md

---
title: Přidání Reactu do existujícího projektu
---

<Intro>

> Pokud máte v plánu přidat interaktivitu do svého existujícího projektu, není nutné celý projekt přepracovat do Reactu. Můžete jednoduše začlenit React mezi již použité technologie a umístit interaktivní React komponenty, kamkoliv je potřeba.

Reviewer: I see great improvement there! Maybe the "do svého existujícího projektu" sounds a bit odd. What about "do existujícího projektu"?

</Intro>

<Note>

**Pro vývoj v lokálním prostředí je nutné mít nainstalovaný [Node.js](https://nodejs.org/en/).** I když máte možnost si [vyzkoušet React](/learn/installation#try-react) online nebo na jednoduché HTML stránce, realita je taková, že většina JavaScriptových nástrojů, které budete pravděpodobně chtít použít pro vývoj, vyžaduje Node.js.

</Note>

> ## Implementace Reactu do adresářové cesty vaší stávající webové stránky {/*using-react-for-an-entire-subroute-of-your-existing-website*/}

Reviewer: I think it is hard to understand. It should be clear that you can use React only on part of the website. I think the "adresářové cesty" is misleading.
```suggestion
## Implementace Reactu do adresářové cesty vaší stávající webové stránky {/*using-react-for-an-entire-subroute-of-your-existing-website*/}
```

> Představme si, že máte existující webovou aplikaci na adrese `domena.cz`, která byla vytvořena za pomocí jiné serverové technologie (např. Rails) a chcete všechny cesty začínající na `domena.cz/obchod/` plně implementovat pomocí Reactu.

Reviewer: Just some minor suggestions. 
```suggestion
Představme si, že máte existující webovou aplikaci na adrese `domena.cz`, která byla vytvořena za pomocí jiné serverové technologie (např. Rails) a chcete, aby všechny cesty začínající s `domena.cz/obchod/` byly plně implementované pomocí Reactu.
```

Reviewer: I think the `example.com` should not be translated.

> Doporučený postup, jak to nastavit:

Reviewer: Maybe?
```suggestion
Zde je náš doporučený postup:
```

1. **Vytvořte část aplikace** pomocí některého z [frameworků založených na Reactu](/learn/start-a-new-react-project).
2. **Nastavte `/obchod` jako *základní cestu*** v konfiguraci vašeho frameworku (zde je návod pro: [Next.js](https://nextjs.org/docs/api-reference/next.config.js/basepath) a [Gatsby](https://www.gatsbyjs.com/docs/how-to/previews-deploys-hosting/path-prefix/)).
3. **Nakonfigurujte svůj server nebo proxy server** tak, aby všechny požadavky na cestu `/obchod/` zpracovávala vaše aplikace v Reactu.

> Takto zajistíte, že část vaší aplikace napsaná Reactu bude moci využít [osvědčené postupy a praktiky](/learn/start-a-new-react-project#can-i-use-react-without-a-framework), které jsou součástí těchto frameworků.

Reviewer: A minor suggestion.
```suggestion
Tímto zajistíte, že část vaší aplikace napsaná Reactu bude využívat [osvědčené postupy a praktiky](/learn/start-a-new-react-project#can-i-use-react-without-a-framework), které jsou součástí těchto frameworků.
```

> Mnoho frameworků založených na Reactu je full-stack, což umožňuje vaší React aplikaci využít server. Tento přístup můžete uplatnit i v případě, kdy nemůžete nebo nechcete na serveru používat JavaScript. V takové situaci místo toho naservírujte export HTML/CSS/JS ([output v `next export`](https://nextjs.org/docs/advanced-features/static-html-export) pro Next.js, default pro Gatsby) na adresářové cestě `/obchod/`.

Reviewer: The word `naservírujete` seems strange to me. What about "zpřístupnite".

## Implementace Reactu do konkrétní části vaší stávající webové stránky {/*using-react-for-a-part-of-your-existing-page*/}

> Předpokládejme, že máte stávající stránku, která byla vytvořena s využitím jiné technologie, ať už serverové (např. Rails) nebo klientské (jako je Backbone), a chcete na ní implementovat interaktivní komponenty React. Tento postup je běžnou formou integrace Reactu. Ve skutečnosti se React takto používal řadu let v Metě!

Reviewer: Is it right to use the word `Metě`?

<html>
  <head><title>My app</title></head>
  <body>
    <!-- Váš stávající obsah stránky (v tomto příkladu bude nahrazen) -->
  </body>
</html>
```

```js index.js active
import { createRoot } from 'react-dom/client';

// Nahradí existující obsah HTML
document.body.innerHTML = '<div id="app"></div>';

// Do něj nyní vykreslíme React komponentu
const root = createRoot(document.getElementById('app'));

> root.render(<h1>Ahoj světe</h1>);

Reviewer: Do we want to translate a text inside the code? We want to translate comments, but I am not sure about this line.

Author: We can decide on this later on. The Spanish translation uses `¡Hola, mundo!`, but we can keep `Hello, world`. I am okay with both.

```

</Sandpack>

> Pokud byl celý obsah stránky nahrazen nápisem "Ahoj světe", vše funguje správně! Čtěte dál.

Reviewer: Just an idea about the form of this sentence.
```suggestion
Pokud byl celý obsah stránky nahrazen nápisem "Ahoj světe", vše funguje správně! Pokračujte ve čtení.
```

Author: Much better, thanks!

<Note>

První integrace modulárního prostředí JavaScriptu do existujícího projektu může působit zastrašujícím dojmem, ale výsledky stojí za to! Pokud narazíte na problémy, využijte naše [komunitní zdroje ](/community) nebo se obrate na [Vite Chat](https://chat.vitejs.dev/).

</Note>

### Krok 2: Vykreslení React komponenty na libovolném místě na stránce {/*step-2-render-react-components-anywhere-on-the-page*/}

V předchozím kroku jste tento kód umístili na začátek hlavního souboru:

```js
import { createRoot } from 'react-dom/client';

// Nahradí existující obsah HTML
document.body.innerHTML = '<div id="app"></div>';

// Do něj nyní vykreslíme React komponentu
const root = createRoot(document.getElementById('app'));
root.render(<h1>Ahoj světe</h1>);
```

> Samozřejmě, v reálném scénáři nebudete chtít existující obsah HTML vymazat!

Reviewer: Maybe different form of the sentence. The "reálném scrénáři" sounds a bit off tome. Maybe "reálném případě" could also fit.
```suggestion
Samozřejmě, v reálu nebudete chtít existující obsah HTML vymazat!
```

Tento kód odstraňte.

> Spíše než vykreslovat celou stránku pomocí Reactu, budete chtít vykreslit React komponenty na specifických místech v HTML. Otevřete svou HTML stránku (nebo serverové šablony, které ji generují) a přidejte například unikátní atribut [`id`](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/id) k libovolnému tagu:

Reviewer: I am very open to discussion, we can translate render to "vykreslit" but also to "rendrovat", which can fit better.

```html
<!-- ... někde v html ... -->
<nav id="navigation"></nav>

> <!-- ... další html ... -->

Reviewer: Maybe?
```suggestion
<!-- ... více html ... -->
```
